Clean/Gadgets/TwoPowerLookup.lean

```diff
@@ -0,0 +1,60 @@
+import Clean.Circuit.Basic
+import Clean.Utils.Field
+
+namespace Gadgets.TwoPowerLookup
+variable {p : ℕ} [Fact (p ≠ 0)] [Fact p.Prime]
+variable [p_large_enough: Fact (p > 512)]
+
+def from_byte_limb {two_power : Fin 9} (x: Fin (2 ^ two_power.val)) : F p :=
```

Comment:
nit: I think `exponent` or `two_exponent` would be better names for `two_power` in this file